cheatsheets/gleam-for-typescript-users.md

```diff
@@ -0,0 +1,826 @@
+const example4 = [1, 2, 3]; // valid, inferred without explicit type annotation
+```
+
+### Gleam
+
+In Gleam type annotations can optionally be given when binding variables.
+
+```gleam
+let some_list: List(Int) = [1, 2, 3]
+let some_string: String = "Foo"
+let another_string = "Bar" // valid, inferred without explicit type annotation
+```
+
+Gleam will check the type annotation to ensure that it matches the type of the
+assigned value. It does not need annotations to type check your code, but you
+may find it useful to annotate variables to hint to the compiler that you want
+a specific type to be inferred.
+## Data types
+
+### Strings
+
+In TypeScript (JavaScript), strings are represented fundamentally as sequences
+of UTF-16 code units. They can be defined with single quotes, double quotes,
+or back-ticks (also known as template strings). Interpolation is supported in
+template literal strings.
+
+In Gleam all strings are UTF-8 encoded binaries. Gleam strings do not allow
+interpolation, yet. Gleam however offers a `string_builder` via its standard
+library for performant string building.
+
+#### TypeScript
+
+```ts
+const firstExample = 'one';
+const secondExample = "two";
+const thirdExample = `three`;
+
+const concat1 = 'one' + 'example';
+const concat2 = "two" + "example";
+const concat3 = `${concat1} ${concat2}`;
+```
+
+
+#### Gleam
+
+```gleam
+let example = "Hellø, world!"
+let another = "Hello" <> ", world!"
+```
+
+### Numbers
+
+#### TypeScript
+
+TypeScript (JavaScript) only has a single number type implemented as a double-precision
+64-bit binary format IEEE 754 value.
+
+```typescript
+1 / 2 // 0.5
+```
+
+#### Gleam
+
+```gleam
+1 / 2 // 0
+1.5 + 10 // Compile time error
+```
+
+You can use the gleam standard library's `int` and `float` modules to convert
+between floats and integers in various ways including `rounding`, `floor`,
+`ceiling` and many more.
+
+Floats in gleam use float-specific operators like `+.` and `/.` instead of sharing `+` and `/`.
+
+```gleam
+1 / 2 // 0
+1.0 /. 2.0 // 0.5
+```
+
+### Arrays and Lists
+
+Arrays in TypeScript are allowed to have values of mixed types, but lists in Gleam can only have a single type.
+
+#### TypeScript
+
+TypeScript has arrays that support mixed types.
+
+```ts
+const arrayOfNumbers = [2, 3, 4];
+const arr2 = [1, "wow", true];
+const arr3: number[] = [1, 2, 3];
+const arr4: Array<number> = [1, 2, 3]; // eqvuialent
+
+const arr5: Array<string|number> = [1, "2", 3]; // Valid. Array of numbers and/or strings
+const arr6: number[] | string[] = [1, 2, 3]; // Valid. Array of numbers OR array of strings
+const arr7: number[] | string[] = ["one", "two", "three"]; // Valid
+const arr6: number[] | string[] = ["one", 2, "three"]; // Invalid
+```
+
+#### Gleam
+
+Gleam has a `cons` operator that works for lists destructuring and
+pattern matching. In Gleam lists are immutable so adding and removing elements
+from the start of a list is highly efficient.
```

Comment:
It might be good to mention that lists are singly-linked so they're only efficient when adding or removing from the start of a list and that elsewhere is less efficient (has to traverse all previous nodes to get to the right one).

Reply:
Has this been added in a local commit and just not been pushed to Github yet?